File: CPP/String/String Matching in an Array.cpp
class Solution {
public:
    bool check(string s, int index,vector<string>words){
        for(int i=0;i<words.size();i++){
          //find() method in C++ returns the position of the first occurrence of the substring, but it returns std::string::npos if the substring is not found. You need to check if the result of find() is not equal to std::string::npos to determine if the substring exists in the other strings
            if(i!=index && words[i].find(s) !=string::npos){
                return true;
            }
        }
        return false;
    }
    vector<string> stringMatching(vector<string>& words) {
        vector<string>ans;
        for(int i=0;i<words.size();i++){
            if(check(words[i],i,words)){
                ans.push_back(words[i]);
            }
        }
        return ans;
    }
};

File: Python/String/String Matching in an Array.py
class Solution:
    def check(self, s:str, index:int, words: List[str])->bool:
        i = 0
        while i<len(words):
            if i!=index and words[i].find(s) !=-1:
              return True
            i=i+1
        return False
    def stringMatching(self, words: List[str]) -> List[str]:
        li=list()
        i=0
        while i<len(words):
            if self.check(words[i],i,words):
                li.append(words[i])
            i=i+1
        return li

############ Methpd 2-------------------
 # def stringMatching(self, words: List[str]) -> List[str]:
 #        ans =set()
 #        for i in range(len(words)):
 #            for j in range(len(words)):
 #                if i!=j and words[i] in words[j]:
 #                    ans.add(words[i])
 #        return list(ans)
